### ISSUE TITLE
AppBundle -> Upload to GPC -> Install -> Run -> Crash(only on master branch)

### ISSUE THREAD

**Issue by worldofsites** [flutter/flutter#33519]
I'm on **master** flutter branch.
I've created an app and successfully tested release version.
```flutter build apk```
All works on emulator and physical device.
Then, I've created appbundle.
```flutter build appbundle```
Uploaded aab file to Google Play Console.
And on internal testing/alpha/beta/prod (same for pre-launch report) next behaviour:
Successfulling install app -> Starting app -> **crash**.

After that, I checked **stable** flutter branch and repeat all the previous steps -> **All works**.
So, something wrong on **master** branch when we creating app bundle.

**Comment by worldofsites** [comment]
App size statistic from GPC:
![image](https://user-images.githubusercontent.com/23352829/58690822-a2c9d900-83b4-11e9-9905-95498180fc5a.png)
0.23 version - bundle created on master branch
0.24.4 version -bundle created on stable branch

**Comment by matthewlloyd** [comment]
I just saw this too and had to roll back the release and go back to using APKs for GPC.

Here are the logs in ADB. The contents of the AAB are below. This was from a clean build.

```
06-02 23:59:58.769 10423 10423 E flutter : [ERROR:flutter/runtime/dart_vm_data.cc(19)] VM snapshot invalid and could not be inferred from settings.
06-02 23:59:58.769 10423 10423 E flutter : [ERROR:flutter/runtime/dart_vm.cc(241)] Could not setup VM data to bootstrap the VM from.
06-02 23:59:58.769 10423 10423 E flutter : [ERROR:flutter/runtime/dart_vm_lifecycle.cc(89)] Could not create Dart VM instance.
06-02 23:59:58.770 10423 10423 F flutter : [FATAL:flutter/shell/common/shell.cc(218)] Check failed: vm. Must be able to initialize the VM.
06-02 23:59:58.770 10423 10423 F libc    : Fatal signal 6 (SIGABRT), code -6 (SI_TKILL) in tid 10423 (com.redacted), pid 10423 (com.redacted)
06-02 23:59:58.799 10482 10482 I crash_dump64: obtaining output fd from tombstoned, type: kDebuggerdTombstone
06-02 23:59:58.799   910   910 I /system/bin/tombstoned: received crash request for pid 10423
06-02 23:59:58.799 10482 10482 I crash_dump64: performing dump of process 10423 (target tid = 10423)
06-02 23:59:58.804 10482 10482 F DEBUG   : *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***
06-02 23:59:58.804 10482 10482 F DEBUG   : Build fingerprint: 'google/blueline/blueline:9/PQ1A.190105.004/5148680:user/release-keys'
06-02 23:59:58.804 10482 10482 F DEBUG   : Revision: 'MP1.0'
06-02 23:59:58.804 10482 10482 F DEBUG   : ABI: 'arm64'
06-02 23:59:58.804 10482 10482 F DEBUG   : pid: 10423, tid: 10423, name: com.redacted  >>> com.redacted <<<
06-02 23:59:58.804 10482 10482 F DEBUG   : signal 6 (SIGABRT), code -6 (SI_TKILL), fault addr --------
06-02 23:59:58.804 10482 10482 F DEBUG   : Abort message: '[FATAL:flutter/shell/common/shell.cc(218)] Check failed: vm. Must be able to initialize the VM.
06-02 23:59:58.804 10482 10482 F DEBUG   : '
06-02 23:59:58.804 10482 10482 F DEBUG   :     x0  0000000000000000  x1  00000000000028b7  x2  0000000000000006  x3  0000000000000008
06-02 23:59:58.804 10482 10482 F DEBUG   :     x4  0080000000000000  x5  0080000000000000  x6  0080000000000000  x7  0000000000008000
06-02 23:59:58.804 10482 10482 F DEBUG   :     x8  0000000000000083  x9  0000007bc17b9168  x10 fffffff87ffffbdf  x11 0000000000000001
06-02 23:59:58.804 10482 10482 F DEBUG   :     x12 0101010101010101  x13 0000000000000030  x14 ffffffffffffffff  x15 0000000000000018
06-02 23:59:58.804 10482 10482 F DEBUG   :     x16 0000007bc17f32c8  x17 0000007bc1730dcc  x18 0000000007121ebc  x19 00000000000028b7
06-02 23:59:58.804 10482 10482 F DEBUG   :     x20 00000000000028b7  x21 0000007fd4e937f8  x22 0000007b3458a1e0  x23 0000007b3780df00
06-02 23:59:58.804 10482 10482 F DEBUG   :     x24 0000007b252a1150  x25 0000000000000000  x26 0000007b3e214ca0  x27 0000000000000003
06-02 23:59:58.804 10482 10482 F DEBUG   :     x28 0000000000000000  x29 0000007fd4e937e0
06-02 23:59:58.804 10482 10482 F DEBUG   :     sp  0000007fd4e937a0  lr  0000007bc172594c  pc  0000007bc1725974
06-02 23:59:58.805 10482 10482 F DEBUG   :
06-02 23:59:58.805 10482 10482 F DEBUG   : backtrace:
06-02 23:59:58.805 10482 10482 F DEBUG   :     #00 pc 0000000000021974  /system/lib64/libc.so (abort+116)
06-02 23:59:58.805 10482 10482 F DEBUG   :     #01 pc 0000000001081644  /data/app/com.redacted-jyED6WnaH1_FjP64ZYaROg==/split_config.arm64_v8a.apk (offset 0x1064000)
```

```
$ unzip -l 200030.aab | grep snapshot
  6255840  12-31-1969 19:00   base/lib/arm64-v8a/lib_isolate_snapshot_data.so
  8444640  12-31-1969 19:00   base/lib/arm64-v8a/lib_isolate_snapshot_instr.so
    32352  12-31-1969 19:00   base/lib/arm64-v8a/lib_vm_snapshot_data.so
    19200  12-31-1969 19:00   base/lib/arm64-v8a/lib_vm_snapshot_instr.so
  5008280  12-31-1969 19:00   base/lib/armeabi-v7a/lib_isolate_snapshot_data.so
  8073232  12-31-1969 19:00   base/lib/armeabi-v7a/lib_isolate_snapshot_instr.so
    23520  12-31-1969 19:00   base/lib/armeabi-v7a/lib_vm_snapshot_data.so
    12640  12-31-1969 19:00   base/lib/armeabi-v7a/lib_vm_snapshot_instr.so

$ unzip -l 200030.aab | grep libflutter
  8473384  12-31-1969 19:00   base/lib/arm64-v8a/libflutter.so
  6048724  12-31-1969 19:00   base/lib/armeabi-v7a/libflutter.so
```

I'm on `master` but am using a detached head at `e1a784ae3` which is from a few days ago (a recent change broke my build).

```
$ flutter doctor
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel unknown, v1.6.6, on Mac OS X 10.14.5 18F132, locale en-US)

[✓] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
[✓] Xcode - develop for iOS and macOS (Xcode 10.2.1)
[✓] iOS tools - develop for iOS devices
[✓] Android Studio (version 3.4)
[✓] IntelliJ IDEA Community Edition (version 2019.1.3)
[✓] VS Code (version 1.33.1)
[✓] Connected device (1 available)

• No issues found!
```

**Comment by matthewlloyd** [comment]
I think this is the same issue as https://github.com/flutter/flutter/issues/33424, which is related to https://github.com/flutter/flutter/issues/31922 and will apparently be fixed in the forthcoming v1.7.0 dev channel release.

**Comment by blasten** [comment]
Dup of #31922.
